nixos/modules/services/networking/wireguard.nix

```diff
@@ -148,6 +148,15 @@ let
         for matching all IPv6 addresses.'';
       };
 
+      allowedIPsAsRoutes = mkOption {
```

Comment:
Seems like this would be better as a per-device option instead of a per-peer option, for simplicity. It's hard to imagine a situation in which you'd want this on for some and off for others, and in the event you want it off, that's probably for everyone, so it might be a bit of a pain to have to add the =false entry to each and every peer, especially if you have quite a few of them.

Reply:
Your comment makes sens. In my scenario I have a bunch of machines which are allowed to originate `::/0` and some that only have a /28 assigned to them. So I would prefer a per peer configuration. Peers with `::/0` are announcing routes to my machine, the ones with static networks wont. So the per-peer option makes sense for me.

I am not in favor of a "global" (per-device) option which could be flipped per-peer. That seems like a bit too much complexity for this kind of thing.

What do you think?

Reply:
You can't give an allowed ips of 0.0.0.0/0 to multiple peers on a single interface, so I'm not totally certain what you mean.

Reply:
I think he talks about routing policies, where some peers are allowed to announce all subnet sizes 
in `::/0` using a dynamic routing protocol and some peers only have static/precalculated routes.

Reply:
I think @zx2c4 is right, it might be more straight forward to only enable/disable that on a per-device level. If you plan to manage some very custom setup you can probably just add some additional services or pre/post exec statements.
I adjusted my PR to do exactly that.